tests/console/journalctl.pm

```diff
@@ -105,23 +114,24 @@ sub run {
     record_soft_failure "bsc#1171858" if (!skip_fss_check && script_run('journalctl --verify --verify-key=`cat /var/tmp/journalctl-setup-keys.txt`') != 0);
     # Basic journalctl tests: Export journalctl with various arguments and ensure they are not empty
     script_run('echo -e "Reboot time:  `cat /var/tmp/reboottime`\nCurrent time: `date -u \'+%F %T\'`"');
-    die "journalctl empty" if check_journal('', "journalctl.txt");
-    die "journalctl of previous boot empty" if !is_public_cloud && check_journal('--boot=-1', "journalctl-1.txt");
+    die "journalctl output is empty!" unless check_journal('', "journalctl.txt");
+    die "journalctl of previous boot empty" if (is_public_cloud || !check_journal('--boot=-1', "journalctl-1.txt"));
     # Note: Detailled error message is "Specifying boot ID or boot offset has no effect, no persistent journal was found."
-    die "no persistent journal was found" if script_run("journalctl --boot=-1 | grep 'no persistent journal was found'") == 0;
-    die "journalctl after reboot empty"   if check_journal('-S "`cat /var/tmp/reboottime`"', "journalctl-after.txt");
+    die "no persistent journal was found" if (script_run("journalctl --boot=-1 | grep 'no persistent journal was found'") == 0);
+    die "journalctl after reboot empty" unless check_journal('-S "`cat /var/tmp/reboottime`"', "journalctl-after.txt");
     if (check_journal('-U "`cat /var/tmp/reboottime`"', "journalctl-before.txt")) {
         # Check for bsc1173856, i.e. the first date in the journal is newer than the last date
         my $awk    = '{print($1 " " $2 " " $3);}';
         my $f_time = script_output("journalctl -q | head -n 1 | awk '$awk'", proceed_on_failure => 1);
         my $l_time = script_output("journalctl -q | tail -n 1 | awk '$awk'", proceed_on_failure => 1);
-        if (str2time($f_time) > str2time($l_time)) {
-            record_soft_failure "bsc#1173856";
-        } else {
-            die "journalctl before reboot empty";
+        record_info 'head time', "$f_time -> " . str2time($f_time);
+        record_info 'tail time', "$l_time -> " . str2time($l_time);
+        if (str2time($f_time) >= str2time($l_time)) {
+            record_soft_failure "bsc#1173856 - journalctl until breaks when time is set back";
+            die "journalctl before reboot empty" unless is_hyperv;
```

Comment:
To exclude the die from `hyperv` is a good idea!

Reply:
Up to this point, I haven't reproduced the weird time shift behaviour. Let's see after more runs. Thanks for your prompt feedback!